lib/std/fs/path.zig

```diff
@@ -1182,3 +1182,69 @@ fn testRelativeWindows(from: []const u8, to: []const u8, expected_output: []cons
     defer testing.allocator.free(result);
     testing.expectEqualSlices(u8, expected_output, result);
 }
+
+/// Returns the extension of the file name (if any).
+/// This function will search for the file extension (separated by a `.`) and will return the text after the `.`.
+/// Files that end with `.` are considered to have no extension, files that start with `.`
+/// Examples:
+/// - `"main.zig"`     ⇒ `"zig"`
+/// - `"src/main.zig"` ⇒ `"zig"`
+/// - `".gitignore"`   ⇒ `null`
+/// - `"keep."`        ⇒ `null`
+/// - `"src.keep.me"`  ⇒ `"me"`
+pub fn extension(path: []const u8) ?[]const u8 {
+    const filename = basename(path);
```

Comment:
What about this?
```zig
const separators = if (builtin.os.tag == .windows) "/\\" else "/";
const index = std.mem.lastIndexOfAny(u8, filename, "." ++ separators) orelse { return null; }
return if (filename[pos] == '.') filename[pos + 1..] else null;
```
The advantage of this over the implementation in this PR is we only search back to the ".", rather than having to go back to the pathname separator first.

Reply:
This will not work for a name `/bar/foo.x/` which will currently (and by design) return `x`

Reply:
EDIT: in my method, will also need to check:
```zig
return if (filename[index] == '.' and index > 0 and !std.fs.path.isSeparator(filename[index - 1])) 
    filename[index + 1..] else null;
```